### PR TITLE
Add charts to stats sections

### LINKED ISSUE #7701: [Analytics Hub] Add charts to the Stats sections

### PR DESCRIPTION
**Pull request by atorresveiga**
⚠️ This PR depends on [this FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2558)

Closes: #7701

### Why
As part of the new Analytics Hub feature, we want to display charts representing the stats behavior by interval as shown in the image below, providing more info and a better experience for merchants.

### Description
The main goal of this PR is to add line charts to the Analytics Revenue and Orders stats cards. 
In order to add the charts, now the `AnalyticRepository` fetches the whole `WCRevenueStatsModel` and not only the totals. This way, we are able to get the period interval lists that we need to draw the charts. 
The `OrderStat` and `RevenueStat` classes were also updated to include the intervals that will be represented by the charts.

The charts are created using the LineChart composable function added in this PR. 
LineChart includes two cases:
1. There is only one value to represent. In this case, the chart is drawn has a single line from start to end.
2. There are multiple values to represent. The common line chart case that is represented as a line joining the different values.

### Testing instructions

1. Open MyStore Screen
2. Tap on See more below the store stats
3. Check that **when** the delta tag is visible, besides the tag, there is a chart matching the tag color (red | green) representing the stat value by interval

### Images/gif

| Light Mode  | Dark Mode |
| ------------- | ------------- |
|  |   |

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/LineChart.kt [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+            )
+        }
+        else -> {
+            MultipleValuesLineChart(
+                info = info,
+                color = color,
+                strokeWidth = strokeWidth,
+                modifier = modifier
+            )
+        }
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/LineChart.kt:48]: LineChart handles 3 main cases:
1. No info data (info is empty) -> No Chart (Maybe a future improvement could be to add an empty state 🤔)
2. List with one info item -> A single line from start to end
3. List with multiple values -> Normal LineChart

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/informationcard/AnalyticsInformationSectionView.kt [hunk 2]
```diff
@@ -53,7 +58,25 @@ internal class AnalyticsInformationSectionView @JvmOverloads constructor(
...
+        binding.cardInformationSectionDeltaTag.isVisible = isDeltaNotNull
+        binding.cardInformationChart.apply {
+            isVisible = isDeltaNotNull
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                val color = if (sectionViewState.delta.greaterThan(0)) {
+                    colorResource(R.color.analytics_delta_positive_color)
+                } else {
+                    colorResource(R.color.analytics_delta_tag_negative_color)
+                }
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/informationcard/AnalyticsInformationSectionView.kt:71]: The chart color depends on the delta value. (Greater than 0 green, else red)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/LineChart.kt [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    color: Color,
+    strokeWidth: Dp,
+    modifier: Modifier = Modifier,
+) {
+    val transparentGraphColor = remember {
+        color.copy(alpha = 0.5f)
+    }
+
+    val higherValue = remember(value) { value + 1f }
+    val lowerValue = remember(value) { value - .15f }
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/LineChart.kt:64]: In the case of a single value chart, the `y` position will be 15% above the bottom of the chart

**Review by nbradbury — APPROVED** [review]
@atorresveiga This all looks good to me! I'll approve this and you can merge when @ThomazFB is done with the FluxC review.

**Comment by ThomazFB** [comment]
@nbradbury I just merged the FluxC side, this PR is good to go!

**Pull request merged**